### PR TITLE
MAINT: Linecheck link fixes

### PR DESCRIPTION
**Pull request by gfyoung**
Title is self-explanatory.  Checked all links with `make linkcheck` and then verified by hand that the identified broken links were actually broken (not always the case).

### CODE REVIEW

File: www/install.rst [hunk 1]
```diff
@@ -96,9 +96,9 @@ and OS-X systems that can be used to install the packages
 one by one. These installers are generally built to be compatible
 with the Python binaries available from python.org.
 
-*  `python <http://python.org/download/>`_
-*  `numpy <http://sourceforge.net/projects/numpy/files/NumPy/>`_
-*  `scipy <http://sourceforge.net/projects/scipy/files/scipy/>`_
+*  `python <https://www.python.org/downloads/>`_
+*  `numpy <https://sourceforge.net/projects/numpy/files/NumPy/>`_
+*  `scipy <https://sourceforge.net/projects/scipy/files/scipy/>`_
```
Comment (ev-br) [inline www/install.rst]: Newer scipy releases are uploaded to Github Releases instead --- https://github.com/scipy/scipy/releases

Comment (gfyoung) [inline www/install.rst]: Done.

File: www/scipylib/building/linux.rst [hunk 2]
```diff
@@ -446,11 +448,13 @@ Note that make ptshared might not work on a single core machine. Note also that
 Step 3: Build UMFPACK (5.5.0) with AMD (2.2.1), UFConfig(3.5.0) and CHOLMOD (1.7.1)
 -----------------------------------------------------------------------------------
 
-NOTE: Dowloading and building the entire `SuiteSparse <http://www.cise.ufl.edu/research/sparse/SuiteSparse/>`__ all at once is easier than the following steps. `SuiteSparse <http://www.cise.ufl.edu/research/sparse/SuiteSparse/>`__ Version 4.0.2 is confirmed to have worked on Xubuntu 12.10 with the latest ATLAS, LAPACK, `NumPy <http://numpy.org>`__ and SciPy as of 2012-10-18.
+NOTE: Downloading and building the entire `SuiteSparse <https://github.com/jluttine/suitesparse>`__ all at once is easier than the following steps. `SuiteSparse <https://github.com/jluttine/suitesparse>`__ Version 4.0.2 is confirmed to have worked on Xubuntu 12.10 with the latest ATLAS, LAPACK, `NumPy <http://www.numpy.org/>`__ and SciPy as of 2012-10-18.
```
Comment (pv) [inline www/scipylib/building/linux.rst]: Those are not the official suitesparse urls

Comment (gfyoung) [inline www/scipylib/building/linux.rst]: Yes, I know, but the original ones are broken, and the README has a link to this page at<a href="http://faculty.cse.tamu.edu/davis/suitesparse.html">here</a>.  However, that page has no link to the GitHub repository.  I thought that directing them to the repository would be more useful.

File: www/scipylib/building/linux.rst [hunk 3]
```diff
@@ -378,11 +380,11 @@ ATLAS
 
 **There is currently no comfortable way to use ATLAS on openSuse.**
 
-The  `build instructions for ATLAS <http://www.scipy.org/Installing_SciPy/Linux#head-89e1f6afaa3314d98a22c79b063cceee2cc6313c>`__  on this page work, but unfortunately the NumPy and SciPy packages don't work with ATLAS. One could build NumPy and SciPy from sources though, and a relatively painless way to do this is the Sage package. (If you know a comfortable way to make ATLAS work on openSuse, please put it here into the Wiki.)
+The  `build instructions for ATLAS <http://www.scipy.org/install.html#head-89e1f6afaa3314d98a22c79b063cceee2cc6313c>`__  on this page work, but unfortunately the NumPy and SciPy packages don't work with ATLAS. One could build NumPy and SciPy from sources though, and a relatively painless way to do this is the Sage package. (If you know a comfortable way to make ATLAS work on openSuse, please put it here into the Wiki.)
```
Comment (pv) [inline www/scipylib/building/linux.rst]: This is a link to a different section on this page, can probably be written as (implicit title target)
```
`build instructions for ATLAS <Building ATLAS>`_
```

File: www/scipylib/building/linux.rst (hunk 2 above)
Comment (gfyoung) [inline www/scipylib/building/linux.rst]: I think what I'll do is reword that section so I can incorporate both links (the official one and the GitHub) into the section.

Comment (pv) [inline www/scipylib/building/linux.rst]: The github one is just some random guy (apologies to jluttine :) not affiliated with the project --- so I wouldn't link to it.

Comment (gfyoung) [inline www/scipylib/building/linux.rst]: Fair enough.  Fixed.

File: www/scipylib/building/linux.rst (hunk 3 above)
Comment (gfyoung) [inline www/scipylib/building/linux.rst]: Ah, good point.  Done.

**Comment by gfyoung** [comment]
Alright, I think I've fixed all of the broken links.  Any broken ones that you may see now when you run `make linkcheck`  can be confirmed by hand that they are not in fact broken AFAICT.  Comments are welcome!  Otherwise, should be good to merge.

**Comment by pv** [comment]
thanks, merged

**Pull request merged**